Fetch/Request.swift

```diff
@@ -0,0 +1,34 @@
+//
+//  Request.swift
+//  Fetch
+//
+//  Created by David Hardiman on 13/02/2016.
+//  Copyright © 2016 David Hardiman. All rights reserved.
+//
+
+import Foundation
+
+/**
+ *  Simple struct representing a network request
+ */
+public struct Request {
```

Comment:
We could just use `NSURLRequest` directly, but it's a bit manky from swift, especially as you have to use the mutable alternative. Seemed nice to hide it behind pure swift.